packages/pipeline-editor/src/PropertiesPanel/index.tsx

```diff
@@ -105,6 +107,9 @@ function PropertiesPanel({
           switch (id) {
             case "browse_file":
               return await onFileRequested?.(data);
+            case "refresh_properties":
+              data.filename = selectedNode.app_data.filename;
+              return await onPropertiesUpdateRequested?.(data);
```

Comment:
I missed this earlier, but this makes me uncomfortable. 
can we do something like this instead:
```ts
return await onPropertiesUpdateRequested?.({
    ...data,
    filename: selectedNode.app_data.filename
});
```

but also, do we need to send back all the properties? Can we just do `onPropertiesUpdateRequested?.(nodeID);` instead?

Reply:
I can make that change

> do we need to send back all the properties

we're not sending back all the properties, for this action `data` is `{ env_vars: items }` as set in `handleRefreshProperties` in `StringArrayControl.tsx`.

Reply:
oh gotcha, do we need to send the `env_vars` data back?

Reply:
I guess we dont need to if I move the merging code to the editor side, I can give that a try

Reply:
update pushed

Reply:
Oh sorry, I misread that. The merging logic should stay on the jupyter side, but the jupyter side should also already have the most up-to-date data, right?